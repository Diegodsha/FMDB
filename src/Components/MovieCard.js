import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Card from '../Styles/Card.module.css';

const MovieCard = ({ movie, button, type }) => (
  <div className={`col-3 col-md-3 col-lg-2 ${Card.width}`}>
    <div className={`card bg-dark text-white ${Card.cardStyled}`}>
      <img
        src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
        className="card-img"
        alt={movie.title}
      />
      <div
        className={`card-img-overlay d-flex align-items-end ${Card.overlay}`}
      >
        {button === 'yes' ? (
          <Link
            to={`details/${type === 'series' ? 'serie' : 'movie'}/${movie.id}`}
            className={`w-100 ${Card.index}`}
          >
            <button
              className={`btn btn-outline-warning w-100 ${Card.cardBtn}`}
              type="button"
            >
              See more
            </button>
          </Link>
        ) : null}
      </div>
    </div>
  </div>
);

MovieCard.defaultProps = {
  button: 'no',
  type: 'movies',
};

MovieCard.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string,
    release_date: PropTypes.string,
    vote_average: PropTypes.number,
    poster_path: PropTypes.string,
    id: PropTypes.number.isRequired,
    name: PropTypes.string,
    first_air_date: PropTypes.string,
  }).isRequired,
  button: PropTypes.string,
  type: PropTypes.string,
};

export default MovieCard;
