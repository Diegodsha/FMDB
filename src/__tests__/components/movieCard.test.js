import renderer from 'react-test-renderer';
import MovieCard from '../../Components/MovieCard';

describe('MovieCard Component', () => {
  test('Renders correctly component', () => {
    const movie = {
      title: 'movie',
      release_date: '2021',
      vote_average: 5,
      poster_path: '/path',
      id: 123,
    };
    const tree = renderer.create(<MovieCard movie={movie} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
