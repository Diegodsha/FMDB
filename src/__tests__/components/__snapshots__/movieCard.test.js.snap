// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Component Renders correctly component 1`] = `
<div
  className="col-3 col-md-3 col-lg-2 width"
>
  <div
    className="card bg-dark text-white cardStyled"
  >
    <img
      alt="movie"
      className="card-img"
      src="https://image.tmdb.org/t/p/w500/path"
    />
    <div
      className="card-img-overlay d-flex align-items-end overlay"
    />
  </div>
</div>
`;
